lib/fs.js

```diff
@@ -2138,3 +2138,24 @@ SyncWriteStream.prototype.destroy = function() {
 };
 
 SyncWriteStream.prototype.destroySoon = SyncWriteStream.prototype.destroy;
+
+fs.mkdtemp = function(template, callback) {
```

Comment:
You can validate the type of `template` here itself, instead of going till C++ layer and failing.